### PR TITLE
[RFC] pass highlight attrs per value and thread-safely to TUI thread

### PR DESCRIPTION
**Pull request by bfredl**
`HlAttrs` is a value type, no need to pass around pointers to it.

Also don't use `syn_id2attr`+`syn_attr2entry` in TUI thread, it is not safe (if TUI changes cursor color the same time as main thread reloads the color scheme, there might be a data race), instead pass the attribute by id like all other attributes now are in UI protocol.

### CODE REVIEW

**Comment by bfredl** [comment]
Added a test for shape is updated also by `hi Cursor guibg=#abcdef` (any group that is used by `guicursor`). So fixes #6591.

File: runtime/doc/ui.txt [hunk 1]
```diff
@@ -89,13 +89,18 @@ Global Events							    *ui-global*
...
 	`name`:		Mode descriptive name.
 	`mouse_shape`:	(To be implemented.)
 
-	Some keys are missing in some modes.
+	Some keys are missing in some modes. The following keys are deprecated:
+
+	`hl_id`:	Cursor highlight group.
+	`hl_lm`:	Cursor highlight group if 'langmap' is active.
+
+    Use `attr_id` and `attr_id_lm` instead.
```
Comment (justinmk) [inline runtime/doc/ui.txt]:
```
Some keys are missing in some modes. 

The following keys are deprecated:

`hl_id`:	Use attr_id instead.
`hl_lm`:	Use attr_id_lm instead.
```

File: src/nvim/cursor_shape.c [hunk 2]
```diff
@@ -64,6 +64,9 @@ Array mode_style_array(void)
       PUT(dic, "blinkoff", INTEGER_OBJ(cur->blinkoff));
       PUT(dic, "hl_id", INTEGER_OBJ(cur->id));
       PUT(dic, "id_lm", INTEGER_OBJ(cur->id_lm));
+      PUT(dic, "attr_id", INTEGER_OBJ(cur->id ? syn_id2attr(cur->id) : 0));
+      PUT(dic, "attr_id_lm", INTEGER_OBJ(cur->id_lm ? syn_id2attr(cur->id_lm)
+                                         : 0));
```
Comment (justinmk) [inline src/nvim/cursor_shape.c]: usually aligned with the `?`

**Review by justinmk — APPROVED** [review]
Awesome!

File: src/nvim/cursor_shape.c (hunk 2 above)
Comment (bfredl) [reply src/nvim/cursor_shape.c]: the linter complained, but guess I could ignore it.

**Review by justinmk — APPROVED** [review]

**Pull request merged**